src/components/TodoHeader/TodoHeader.tsx

```diff
@@ -0,0 +1,98 @@
+import { useEffect, useRef } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  activeTodosCount: number,
+  onSubmit: (todo: Todo) => void,
+  todo?: Todo | null,
+  userId: number,
+  isLoading: boolean,
+  errorMessage: string,
+  request: boolean;
+  setErrorMessage: (errorMessage: string) => void,
+  title: string;
+  setTitle: (title: string) => void,
+  isAllCompleted: boolean;
+  onToggleAll: () => void;
+};
+
+export const TodoHeader: React.FC<Props> = ({
+  activeTodosCount,
+  onSubmit,
+  todo,
+  userId,
+  errorMessage,
+  setErrorMessage,
+  request,
+  title,
+  setTitle,
+  isAllCompleted,
+  onToggleAll,
+}) => {
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [activeTodosCount, request]);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const newTitle = event.target.value;
+
+    setTitle(newTitle);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    setErrorMessage('');
+
+    if (!title.trimStart()) {
+      setErrorMessage('Title should not be empty');
+
+      return;
+    }
+
+    const id = todo?.id || 0;
+    const tempTodo: Omit<Todo, 'id'> = {
+      title: title.trim(),
+      completed: false,
+      userId,
+    };
+
+    onSubmit({ id, ...tempTodo });
```

Comment:
It looks strange, you omit `id` but pass with `id` to the `onSubmit` and I don't see any other places where you use `tempTodo` here